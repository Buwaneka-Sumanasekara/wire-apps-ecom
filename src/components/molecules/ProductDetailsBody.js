import React, { useState } from "react"
import { Text, View } from "react-native"
import ProductBasicInfo from "../atoms/ProductBasicInfo"
import withPaddings from "../../templates/withPaddings"
import ProductSizes from "../atoms/ProductSizes"
import { useChangeProductSize } from "../../context/ProductContext"
import QtyInput from "../atoms/QtyInput"
import AddToCartButton from "../atoms/AddToCartButton"
import { useAddItemToCart } from "../../hooks/useCart"
import { showMessage } from "react-native-flash-message"



const ProductDetailsBody = (props) => {
    const [qty,setQty]=useState(1)
    const {onChangeSize,selectedSize}=useChangeProductSize();
    const {addToCart}=useAddItemToCart(()=>showMessage({message:"Success",description:"Item added to the cart",type:"success"}));

    return (
        <View className={"pt-3 pb-32"}>
            <ProductBasicInfo {...props} />
            <ProductSizes {...props} selectedSize={selectedSize} onChange={onChangeSize} />
            <View className={"mt-5 flex flex-row"}>
                <QtyInput onChangeQty={setQty} qty={qty} />
                <AddToCartButton  isDisabled={!selectedSize} onPress={()=>{
                    addToCart(props,selectedSize,qty)
                }}/>
            </View>
            <View className={"mt-5 w-fit"}>
                <Text>{props.description}</Text>
            </View>
        </View>
    )
}

export default withPaddings(ProductDetailsBody)