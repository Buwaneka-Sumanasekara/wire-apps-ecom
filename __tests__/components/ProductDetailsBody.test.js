import React from 'react';
import renderer from 'react-test-renderer';
import ProductDetailsBody from '../../src/components/molecules/ProductDetailsBody';
import productJson from '../data/product.json';

describe('<ProductDetailsBody/>', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<ProductDetailsBody item={productJson} />).toJSON()
        expect(tree).toMatchSnapshot()
    });
});

