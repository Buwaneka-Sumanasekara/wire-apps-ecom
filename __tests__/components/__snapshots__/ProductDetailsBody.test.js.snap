// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailsBody/> rendered sucessfully 1`] = `
<View
  style={
    [
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "paddingTop": 12,
        },
        {
          "paddingBottom": 128,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "width": "66.666667%",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#475569",
              },
              {
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Nike
        </Text>
        <Text
          style={
            [
              {
                "fontWeight": "700",
              },
              {
                "fontSize": 18,
                "lineHeight": 28,
              },
            ]
          }
        >
          Nike Air Relentless 4 Mens Running Shoes
        </Text>
      </View>
      <View
        style={
          [
            {
              "width": "33.333333%",
            },
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "700",
              },
              {
                "fontSize": 18,
                "lineHeight": 28,
              },
            ]
          }
        >
          45 GBP
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 12,
          },
        ]
      }
    >
      <Text>
        Choose size
      </Text>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
            {
              "flexWrap": "wrap",
            },
            {
              "marginLeft": -8,
              "marginTop": -8,
            },
            {
              "width": 750,
            },
            {
              "marginTop": 4,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f1f5f9",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "marginLeft": 8,
              "marginTop": 8,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000",
                },
              ]
            }
          >
            8
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f1f5f9",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "marginLeft": 8,
              "marginTop": 8,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000",
                },
              ]
            }
          >
            9
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f1f5f9",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "marginLeft": 8,
              "marginTop": 8,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000",
                },
              ]
            }
          >
            10
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#f1f5f9",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "marginLeft": 8,
              "marginTop": 8,
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#000",
                },
              ]
            }
          >
            11
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 20,
          },
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#000",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <Text />
        </View>
        <TextInput
          editable={false}
          style={
            [
              {
                "borderBottomWidth": 1,
                "borderLeftWidth": 1,
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              {
                "borderBottomColor": "#d1d5db",
                "borderLeftColor": "#d1d5db",
                "borderRightColor": "#d1d5db",
                "borderTopColor": "#d1d5db",
              },
              {
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              },
              {
                "width": 40,
              },
              {
                "textAlign": "center",
              },
              {
                "alignItems": "center",
              },
            ]
          }
          value="1"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#000",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        style={
          [
            {
              "paddingRight": 8,
            },
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#030303",
              },
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              },
              {
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
              },
            ]
          }
        >
          <Text />
          <Text
            style={
              [
                {
                  "color": "#64748b",
                },
                {
                  "marginLeft": 4,
                },
              ]
            }
          >
            Add To Cart
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 20,
          },
        ]
      }
    >
      <Text>
        Hit the tracks in these Nike Air Relentless 4 featuring flexible forefoot sole and Reslon midsole underfoot cushioning for superior comfort at each step. The ridged outsole ensures excellent traction while the cushioned ankle collar and the anatomically shaped insole guarantee great support for the whole foot. The mesh upper panels provide breathability and airflow within the shoe.
      </Text>
    </View>
  </View>
</View>
`;
